pyiron_contrib/generic/s3io.py

```diff
@@ -0,0 +1,564 @@
+        if metadata is None:
+            metadata = {}
+
+        if isinstance(files, str):
+            files = [files]
+
+        for file in files:
+            [_, filename] = os.path.split(file)
+
+            self._bucket.upload_file(
+                file,
+                self._bucket_path + filename,
+                {"Metadata": metadata}
+            )
+
+    def download(self, files, targetpath="."):
+        """
+        Download files from current group to local file system (current directory is default)
+
+        Arguments:
+            files (list/str): List of filenames in the S3 object store.
+            targetpath (str): Path in the local data system, to which the files should be downloaded.
+        """
+        if not os.path.exists(targetpath):
+            os.mkdir(targetpath)
+        if isinstance(files, str):
+            files = [files]
+
+        for f in files:
+            filepath = os.path.join(targetpath, f.split("/")[-1])
+            print(filepath)
+            self._bucket.download_file(self._bucket_path + f, filepath)
+
+    def get_metadata(self, file):
+        """
+        Returns the metadata of a file.
+
+        Args:
+            file (str): path to a file of the bucket.
+        Returns:
+             dict: metadata field associated with the file.
+        """
+        file = self._to_abs_bucketpath(file)
+        return self._bucket.Object(file).metadata
+
+    def _s3io_object(self, file):
+        """
+        Returns an object with access to the S3 object store which can be downloaded via .get()
+
+        Args:
+            file (str): path to a file of the bucket.
+        Returns:
+        """
+        file = self._to_abs_bucketpath(file)
+        s3object = S3FileData(s3obj=self._bucket.Object(file))
+        return s3object
+
+    def get(self, file):
+        """
+        Returns a s3.Object containing the requested file.
+
+        Args:
+            file(str): a path like string.
+        Returns:
+             Object containing a file.
+        """
+        file = self._to_abs_bucketpath(file)
+        return self._bucket.Object(file).get()
+
+    def put(self, data_obj, path=None, metadata=None):
+        """
+            Upload a data_obj to the current group/ the provided path.
+
+            Args:
+                data_obj(:class:`pyiron_base.generic.filedata.FileData`): data object to upload the data from.
+                path(str/None):
+                metadata(dict/None): metadata to be used (has to be a dictionary of type {"string": "string, }).
+                      Provided metadata overwrites the one possibly present in the data object.
+        """
+        if self.is_dir(path):
+            path = self._to_abs_bucketpath(path)
+        else:
+            raise ValueError("No valid path specified!")
+        if path[-1] != '/':
+            path = path + '/'
+        path = path + data_obj.filename
+
+        data = data_obj.data()
+        if metadata is None:
```

Comment:
I'm not a huge fan of the double-if implicit "if it's *still* `None`". How do you feel about this two-liner? (I tried a one-liner starting with `metadata if metadata is not None else...` and it runs fine, but I think the line length is too long.)

```suggestion
        if metadata is None:
            metadata = data_obj.metadata if data_obj.metadata is not None else {}
```